mysys/my_atomic_writes.c

```diff
@@ -358,12 +376,68 @@ static my_bool sfx_has_atomic_write(File file, int page_size)
   }
   return 0;
 }
+
+static my_bool sfx_dev_could_disable_punch_hole(struct sfx_dev *dev, File file)
+{
+  int result = 0;
+
+  if (dev->disable_punch_hole == SFX_UNKNOWN_PUNCH_HOLE_YET)
+  {
+    int fd= open(dev->dev_name, 0);
+    if (fd < 0)
+    {
+      perror("open() failed!");
+      dev->disable_punch_hole= 0;                      /* Don't try again */
+      return FALSE;
+    }
```

Comment:
This code was originally copied from the Shannon detection code. It is not very useful, because neither `dev->dev_name` nor the impact of the error is mentioned.